src/API/Reports/DataStore.php

```diff
@@ -440,7 +440,15 @@ protected function cast_numbers( $array ) {
 	 * @return string
 	 */
 	protected function selected_columns( $query_args ) {
-		$selections = $this->report_columns;
+		/**
+		 * Filter the report columns before retrieving data.
+		 *
+		 * Allows filtering of the report columns included in reports.
+		 *
+		 * @param array $report_columns Associative list of columns.
+		 * @param array $query_args     The original arguments for the request.
+		 */
+		$selections = apply_filters( 'wc_admin_reports_selected_columns', $this->report_columns, $query_args );
```

Comment:
Nice work on this @rrennick.

I think we're going to need to add more context to this filter (and probably to the DataStore classes as well) in order for developers to reliably determine which report they're modifying.

Though it's expanding the scope quite a bit, I think this PR is the place to do it. What do you think about adding `$report_type` to all of the report `DataStore`s and including it in the `DataStoreInterface` and the filter names?

We also need to add a filter for the `$column_types`.

Reply:
@jeffstieler Thanks for the review 💯 We discussed #2714 and #1320 at our team meetings last week and decided to do the refactor so that these filters provide a more consistent interface across data stores (including the ones you noted in your feedback).